reducer/src/main/java/com/graphicsfuzz/reducer/reductionopportunities/StmtReductionOpportunity.java

```diff
@@ -49,10 +53,26 @@ public void applyReductionImpl() {
 
   @Override
   public boolean preconditionHolds() {
-    if (!blockStmt.getStmts().contains(child)) {
+    if (!blockStmt.getStmts().contains(childOfBlockStmt)) {
       // Some other reduction opportunity must have removed the statement already
       return false;
     }
+    if (blockStmt.getStmt(blockStmt.getNumStmts() - 1) == childOfBlockStmt
```

Comment:
Could you add comments showing some glsl code and what this (Java) code handles?